src/components/App/App.tsx

```diff
@@ -598,6 +600,39 @@ export default class App extends React.Component<AppProps, AppState> {
             : this.state.unreadCount + 1,
       });
     });
+    socket.on('REC:addReaction', (data: Reaction) => {
+      const { chat } = this.state;
+      const msgIndex = chat.findIndex(
+        (m) => m.id === data.msgId && m.timestamp === data.msgTimestamp
+      );
+      if (msgIndex === -1) {
+        return;
+      }
+      const msg = chat[msgIndex];
+      msg.reactions = msg.reactions || {};
+      msg.reactions[data.value] = msg.reactions[data.value] || [];
+      msg.reactions[data.value].push(data.user);
+      this.setState({ chat }, () => {
+        // if we add a reaction to the last message we need to scroll down
+        // or else the reaction icon might be hidden
+        if (msgIndex === chat.length - 1) {
+          this.chatRef.current?.scrollToBottom();
```

Comment:
Could this cause a problem if the user is scrolled up (e.g. to look at history) and someone sends a react? Maybe we can handle this in the chat's existing scroll to bottom logic, or reuse it (since it handles if the user is scrolled up?)

Reply:
Yes, it would be a problem. This should fix it.